### ISSUE TITLE
Dashboards: Adhoc filter does not work with double quotes

### ISSUE THREAD

**Issue by pavel-mokrushin** [grafana/grafana#72008]
### What happened?

When I click on "filter for value" on a text value that contains double quotes, it shows nothing. If I escape these quotes in ad hoc filter, then the data is displayed.

https://github.com/grafana/grafana/assets/82204499/7dd94bd3-f25c-4f52-9da5-c6cf9f24ad60

I noticed that Grafana versions 9.1 and 10.0 use different queries for elasticsearch when using ad hoc filters.
It differs in that a separate parameter was used earlier `\"must\":[{\"match_phrase\":` But now the ad hoc filter is just added to the request
For 9.1:
```
data:"{\"search_type\":\"query_then_fetch\",\"ignore_unavailable\":true,\"index\":\"rx-log-error*\"}
{\"size\":0,\"query\":{\"bool\":{\"filter\":[{\"range\":{\"@timestamp\":{\"gte\":1689845161005,\"lte\":1689848761005,\"format\":\"epoch_millis\"}}},{\"query_string\":{\"analyze_wildcard\":true,\"query\":\"logLevel: Error AND exceptionType: (\\\"critical\\\" OR \\\"noncritical\\\" OR \\\"unknown\\\")\"}}],\"must\":[{\"match_phrase\":{\"genericMessage\":{\"query\":\"Нет прав на выполнение операции \\\"{S}\\\" на \\\"{S}\\\" с ИД={N}.\"}}}]}},\"aggs\":{\"3\":{\"terms\":{\"field\":\"genericMessage\",\"size\":500,\"order\":{\"_count\":\"desc\"},\"min_doc_count\":1,\"missing\":\"Empty\"},\"aggs\":{}}}}
"
```
For 10.0:
```
data:"{\"search_type\":\"query_then_fetch\",\"ignore_unavailable\":true,\"index\":\"rx-log*\"}
{\"size\":0,\"query\":{\"bool\":{\"filter\":[{\"range\":{\"@timestamp\":{\"gte\":1689762519433,\"lte\":1689848919433,\"format\":\"epoch_millis\"}}},{\"query_string\":{\"analyze_wildcard\":true,\"query\":\"logLevel: Error AND genericMessage:\\\"Cannot unlock: the item \\\"{S}\\\" with ID {N} was not locked.\\\"\"}}]}},\"aggs\":{\"3\":{\"terms\":{\"field\":\"genericMessage\",\"size\":500,\"order\":{\"_count\":\"desc\"},\"min_doc_count\":1,\"missing\":\"Empty\"},\"aggs\":{}}}}
"
```

### What did you expect to happen?

The data should be filtered, including the value with double quotes

### Did this work before?

It worked in grafana versions 9.0.x, 9.1.x, 9.2.x
9.3 and higher does not work

### How do we reproduce it?

1. There must be a value with text data that contains double quotes
2. On the table panel, click "filter for value" or "filter out value".

### Environment (with versions)?

Grafana: 10.0.2
OS: Linux Ubuntu (Grafana-oss in docker)
Browser: Chrome 114

### Grafana platform?

Docker

### Datasource(s)?

Elasticsearch 7.16

**Comment by mellieA** [comment]
Similar to https://github.com/grafana/grafana/issues/70230

**Comment by ivanahuckova** [comment]
@matyax was this fixed by https://github.com/grafana/grafana/issues/70230?

**Comment by matyax** [comment]
@ivanahuckova yes! Should be already fixed.

**Comment by matyax** [comment]
@ivanahuckova apologies, amending my recent comment. Actually https://github.com/grafana/grafana/pull/71954 would fix it, since there I'm using `lucene.escape`, where in #70230 we focused on fields with colons `:`. I'll double check that https://github.com/grafana/grafana/pull/71954 fixes this.

**Comment by matyax** [comment]
https://github.com/grafana/grafana/pull/71954/commits/4fb60131f74b5a50433dd9f848e30433dca57763 will take care of fixing this.

**Comment by pavel-mokrushin** [comment]
Tell me, please, in which version of Grafana will this fix be?

**Comment by matyax** [comment]
10.1 which is going to be released soon @pavel-mokrushin

**Comment by pavel-mokrushin** [comment]
Thanks for the response and quick fix

**Comment by pavel-mokrushin** [comment]
Hi. I installed Grafana version 10.1.0 and the problem remained. The fix didn't make it to version 10.1?
